src/test/java/jenkins/scm/impl/mock/MockSCMSource.java

```diff
@@ -72,28 +75,63 @@
     private final List<SCMSourceTrait> traits;
     private transient MockSCMController controller;
 
-    @DataBoundConstructor
+    @Restricted(NoExternalUse.class)
```

Comment:
Completely ignored in test sources.

We make no guarantees about either binary or source compatibility of test utilities, so either leave it simply `@Deprecated`, if it can continue to work as before (which I think it can), or delete it if it cannot.